typescript/sdk/src/token/adapters/EvmERC721TokenAdapter.ts

```diff
@@ -0,0 +1,157 @@
+import { PopulatedTransaction } from '@ethersproject/contracts';
+import { BigNumberish, ethers } from 'ethers';
+
+import {
+  ERC721,
+  ERC721__factory,
+  HypERC721Collateral,
+  HypERC721Collateral__factory,
+} from '@hyperlane-xyz/core';
+import type {
+  InterchainGasQuote,
+  Token,
+  TransferRemoteParams,
+} from '@hyperlane-xyz/sdk';
+import {
+  EvmHypSyntheticAdapter,
+  EvmTokenAdapter,
+  IHypTokenAdapter,
+  MultiProtocolProvider,
+  TransferParams,
+} from '@hyperlane-xyz/sdk';
+
+// Interacts with HypCollateral contracts
+export class EvmHypERC721CollateralAdapter
+  extends EvmHypSyntheticAdapter
+  implements IHypTokenAdapter<PopulatedTransaction>
+{
```

Comment:
afaict this is basically consistent with ERC20 except for the approve interface on ERC721?

Reply:
Yup think it's just the approve fn